test/new-e2e/tests/process/testing.go

```diff
@@ -207,6 +208,21 @@ func filterProcess(name string, processes []*agentmodel.Process) []*agentmodel.P
 	return found
 }
 
+// filterPayloadsSince filters payloads that were collected since the given time. Also returns the
+// time of the last payload.
+func filterPayloadsSince(payloads []*aggregator.ProcessPayload, since time.Time) ([]*aggregator.ProcessPayload, time.Time) {
+	var filtered []*aggregator.ProcessPayload
+	lastCollectedTime := since
+	for _, payload := range payloads {
+		if payload.GetCollectedTime().After(since) {
+			filtered = append(filtered, payload)
+		}
+		lastCollectedTime = payload.GetCollectedTime()
+	}
+
+	return filtered, lastCollectedTime
+}
```

Comment:
With current time between assert loops, wouldn't we expect this to at most return 1 payload? Do we think thats sufficient for the test?

Also why do you think its best to base it on time, rather than returning the latest payloads?

Reply:
I've updated the test to just inspect the last payload to keep it simple.